Lib/statistics.py

```diff
@@ -304,6 +304,19 @@ def _fail_neg(values, errmsg='negative value'):
             raise StatisticsError(errmsg)
         yield x
 
+def _isqrt_frac_rto(n, m):
+    'Square root of n/m, rounded to the nearest integer using round-to-odd.'
+    a = math.isqrt(n*m) // m
+    return a | (a*a*m != n)
+
+def _sqrt_frac(n, m):
+    'Square root of n/m as a float, correctly rounded.'
```

Comment:
Sometimes, I wish that we would have a math.py module (Python module) to put code like that there :-) Even if it remains private.

Do you think that i would be possible to write unit tests on this function? I'm asking because we might get different results depending on the compiler / platform / float implementation.

Reply:
Added *test_sqrt_frac()* for this helper function.  And if I know Mark, he will likely write a proof for it as well.  Likely it will be platform independent because it uses integer math until the final convert to float step.  Am running the bot fleet as an extra check.

Reply:
Right, the only part that involves floats is the final step, where we rely on int->float conversion being correctly rounded (in the first branch) and int/int true division being correctly rounded (in the second branch). Both those properties should be true for CPython, though I guess that's technically an implementation detail specific to CPython, and not a promise of Python-the-language.